### PR TITLE
hide comments for non-members

### PR DESCRIPTION
**Pull request by michaelbeil**
this is from @ipokkel - https://gist.github.com/ipokkel/c5dd3b0fe720b3a26aab902928abd224

### CODE REVIEW

File: misc/hide-comments-for-non-members.php [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+ 	}
+ 	return $comments;
+ }
+ add_filter( 'comments_array', 'pmpro_hide_comments_for_non_members' );
+
+ function pmpro_non_member_text_before_comment_form() {
+ 	$pmpro_comments_non_member_text = pmpro_get_no_access_message( null, null, null );
+ 	$pmpro_comments_non_member_text = str_replace( __( 'This content is', 'pmpro-comments' ), __( 'Comments are', 'pmpro-comments' ), $pmpro_comments_non_member_text );
+
+ 	echo wp_kses_post( $pmpro_comments_non_member_text );
```
Comment (dparker1005) [inline misc/hide-comments-for-non-members.php]:
```suggestion
 	echo wp_kses_post( $pmpro_comments_non_member_text );
 }
```

**Review by dparker1005 — COMMENTED** [review]
Adding closing bracket

**Comment by dparker1005** [comment]
This code does not hide the comment form for non-members, only shows a "no access" message before it. Should we be hiding the comment form as well?

**Comment by michaelbeil** [comment]
yes, it should be hiding the comments. this was taken from [trello](https://trello.com/c/WPiOlpdv/358-hide-comments-on-public-posts) for off-support-work (untested, as i thought it did exactly that).

**Comment by andrewlimaza** [comment]
I had to redo this due to a bunch of merge conflicts. I refactored it slightly.

**Pull request closed without merging**